consensus/src/aux_schema.rs

```diff
@@ -43,21 +43,30 @@ pub fn block_hash_key(ethereum_block_hash: H256) -> Vec<u8> {
 pub fn load_block_hash<Block: BlockT, B: AuxStore>(
 	backend: &B,
 	hash: H256,
-) -> ClientResult<Option<Block::Hash>> {
+) -> ClientResult<Option<Vec<Block::Hash>>> {
 	let key = block_hash_key(hash);
 	load_decode(backend, &key)
 }
 
 /// Update Aux block hash.
-pub fn write_block_hash<Hash: Encode, F, R>(
+pub fn write_block_hash<Hash: Encode + Decode, F, R, Backend: AuxStore>(
+	client: &Backend,
 	ethereum_hash: H256,
 	block_hash: Hash,
 	write_aux: F,
 ) -> R where
 	F: FnOnce(&[(&[u8], &[u8])]) -> R,
 {
 	let key = block_hash_key(ethereum_hash);
-	write_aux(&[(&key, &block_hash.encode()[..])])
+
+	let mut data: Vec<Hash> = match load_decode(client, &key)
```

Comment:
Not a big fan of having to read the database to set the aux. I'd prefer a merge strategy, but I suspect it is not an easy one to do, so I'm fine with the current implementation

Reply:
I cannot think a way of mimicking SQL merges in a encoded key-value pair database. Afaik if we need to mantain the vector updated, we need to read, update and put it.

Reply:
Well, you could simply replicate the same behavior that you are doing here, but when the database is actually updated, not within the block structure. But as I said, it would require significant changes and probably would not be a desired effect of reusing the same key in the database